### PR TITLE
Add audit job to CI pipeline

### PR DESCRIPTION
**Pull request by jmcph4**
## 📝 Summary

 - Add `cargo audit` invocation to `Makefile` as `make audit`
 - Add `make audit` to existing CI pipeline

## 💡 Motivation and Context

See #79 for context.

---

## ✅ I have completed the following steps:

* [x] Run `make lint`
* [x] Run `make test`
* [x] ~~Added tests (if applicable)~~

### CODE REVIEW

**Review by dvush — APPROVED** [review]

**Comment by metachris** [comment]
This is great, thanks! 

Merging this as-is would make the checks CI step fail. Are the audit issues easy to resolve?

**Comment by jmcph4** [comment]
> Merging this as-is would make the checks CI step fail.

Yeah I knew this would break CI but wanted to file a PR regardless; I agree that it shouldn't be merged in prior to actual fixes.

> Are the audit issues easy to resolve?

It's essentially blocked by two things:

 - Updating our Reth dependencies (#51)
 - Merging in dvush/mempool-dumpster-rs#2, which will fix the same issues upstream. Note that after this occurs, we can either
   - Publish a new crate version for `mempool-dumpster-rs`, or
   - Change the dependency in `rbuilder` to point to the Git repo directly

Personally, I'm leaning towards the latter resolution.

**Comment by metachris** [comment]
After some debate, we prefer not to merge this because it will lead to lots of false positives and block merging PRs.

Thanks for the contribution, looking forward to your next one <3

**Pull request closed without merging**
